src/scene/update.js

```diff
@@ -0,0 +1,87 @@
+import Phaser from 'phaser'
+import {CONFIG} from '../config'
+import {WORLD} from '../world'
+import {Circle, ephemeraInBounds, throtLog} from '../util'
+import {fireBullet} from '../ephemera/bullets'
+import {addPlusOne} from '../ephemera/powerups'
+import {updateEphemeraPost, updateEphemeraPre, updatePre} from '../features'
+// const {Geom} = Phaser
+// const {Circle} = Geom
+
+export function update() {
+  WORLD.graphics.clear()
+  const {acceleration, colors} = CONFIG
+  const {yellow, white} = colors
+  const {inputs, ephemera = {}} = WORLD
+  const {powerups, bullets} = ephemera
+  const {left, right, up, down, space} = inputs
+
+  // We also tried player.physics.setVelocity(0, 0) here instead of friction
+
+  if (left.isDown) {
+    WORLD.player.physics.setVelocityX(-acceleration) // See also .setAcceleration
+  } else if (right.isDown) {
+    WORLD.player.physics.setVelocityX(acceleration)
+  }
+
+  // Not else if here so they can move left & up at the same time, etc.
+  if (up.isDown) {
+    WORLD.player.physics.setVelocityY(-acceleration)
+  // this seems existential
+  } else if (down.isDown) {
+    WORLD.player.physics.setVelocityY(acceleration)
+  }
+
+  // updatePre(this.add.text.bind(this))
+  // console.log(`this`, this)
+
+  if (!WORLD.powerupIntervalForPlusOne && WORLD.ephemera.powerups.length === 0) {
+  // if (alt.isDown) {
+    const addOne = addPlusOne()
+    WORLD.powerupIntervalForPlusOne = addOne
+    addOne.start()
+    setTimeout(() => {
+      delete WORLD.powerupIntervalForPlusOne
+      addOne.stop()
+    }, addOne.lifetime)
+  }
+  // TODO: make this re-usable as a render method instead of this hardcoded
```

Comment:
I'd like to be able to do `powerups.map(addOne.render)` (from inside the block above) but I tried variations on that to no success.